src/language/api/verificationDiagnostics.ts

```diff
@@ -0,0 +1,85 @@
+import { Position } from 'vscode';
+import { Diagnostic, DocumentUri, Range, integer } from 'vscode-languageclient';
+
+export interface IVerificationDiagnosticsParams {
+  uri: DocumentUri;
+  version?: integer;
+  diagnostics: Diagnostic[];
+  perNodeDiagnostic: NodeDiagnostic[];
+  diagnosticsAreResolutionErrors: boolean;
+  perLineDiagnostic: LineVerificationStatus[];
+}
+
+export interface NodeDiagnostic {
+  displayName: string;
+  identifier: string;
+  position: Position;
+  started: boolean;
+  finished: boolean;
+  startTime: integer;
+  endTime: integer;
+  timeSpent: integer;
+  resourceCount: integer;
+  range: Range;
+  children: NodeDiagnostic[];
+  statusCurrent: CurrentStatus;
+  statusVerification: VerificationStatus;
+  relatedRanges: Range[];
+  immediatelyRelatedRanges?: Range[];
+  dynamicallyRelatedRanges?: Range[];
+}
+
+export enum LineVerificationStatus {
```

Comment:
Do these enumeration numbers come from boogie? I suggest to include it in the type-name or at least in a comment to make this clear.

Reply:
I'm adding this comment:
```
// Except for cosmetics, this enumeration is a copy-paste from Dafny's
// Source/DafnyLanguageServer/Workspace/Notifications/VerificationDiagnosticsParams.cs
```

Reply:
Ah that's not necessary since these modules are supposed to represent the language server's public API. I was wondering if this was a custom enumeration or one that basically represents the states observable within boogie since it's a rather huge enumeration. But I suppose these are all states you're setting manually.